samza-core/src/main/java/org/apache/samza/table/remote/RecordReader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.operators.functions.ClosableFunction;
+import org.apache.samza.operators.functions.InitableFunction;
+
+
+/**
+ * Interface for the read accessors to a specific remote store. Implementation
+ * can override init(config, context) and close() for initialization and cleanup.
+ * Any non-serializable states, eg. network connection, should be marked as
+ * transient and recreated by overriding readObject().
+ * @param <K>
+ * @param <V>
+ */
+@InterfaceStability.Unstable
+public interface RecordReader<K, V> extends Serializable, InitableFunction, ClosableFunction {
```

Comment:
I thought RecordReader/Writer are conceptually functions so it makes sense to reuse the abstractions, which is the InitableFunction and ClosableFunction. didn't go with java.io.Closeable because we already have ClosableFunction in Samza. 

Does this make sense?